e2e/cypress/integration/integrations/zapier/oauth_spec.js

```diff
@@ -0,0 +1,440 @@
+                    cy.contains('*******************').should('exist');
+
+                    // * Copy button should not exist
+                    cy.get('.fa-copy').should('not.exist');
+                });
+
+                // # Show secret
+                cy.findByText('Show Secret').click();
+
+                // * Show secret text should have changed to Hide Secret
+                cy.findByText('Hide Secret').should('exist');
+                cy.findByText('Show Secret').should('not.exist');
+
+                cy.contains('.item-details__token', 'Client Secret').within(() => {
+                    // * Token should not be obscured
+                    cy.contains('*******************').should('not.exist');
+
+                    // * Copy button should exist
+                    cy.get('.fa-copy').should('exist');
+                });
+            });
+        });
+    });
+
+    it('MM-T648 Part 1 - OAuth 2.0 Zapier tests', () => {
+        cy.apiLogin(user1);
+        cy.visit(testChannelUrl1);
+
+        // # Navigate to OAuthApps in integrations menu
+        cy.get('#headerInfo').click();
+        cy.get('#integrations').click();
+        cy.get('#oauthApps').click();
+
+        // # Click on the Add button
+        cy.get('#addOauthApp').click();
+
+        // # Fill all fields
+        cy.get('#name').type('Test');
+        cy.get('#description').type('Test');
+        cy.get('#homepage').type('https://www.test.com/');
+        cy.get('#callbackUrls').type(getWebhookBaseUrl() + '/complete_oauth');
+
+        // # Save
+        cy.get('#saveOauthApp').click();
+
+        // * Copy button should be visible
+        cy.get('.fa-copy').should('exist');
+
+        // # Store client ID
+        cy.findByText('Client ID').parent().invoke('text').then((text) => {
+            cy.wrap(text.substring(11)).as('clientID');
+        });
+
+        // # Store client secret
+        cy.findByText('Client Secret').parent().invoke('text').then((text) => {
+            cy.wrap(text.substring(15)).as('clientSecret');
+        });
+
+        cy.get('@clientID').then((clientID) => {
+            oauthClientID = clientID;
+            cy.get('@clientSecret').then((clientSecret) => {
+                oauthClientSecret = clientSecret;
+
+                // # Send credentials
+                cy.postIncomingWebhook({url: getWebhookBaseUrl() + '/send_oauth_credentials', data: {appID: clientID, appSecret: clientSecret}});
+            });
+        });
+
+        // # Click Done
+        cy.get('#doneButton').click();
+    });
+
+    it('Aux - Exchange tokens', () => {
+        cy.apiLogin(user1);
+
+        // # Visit the webhook url to start the OAuth handshake
+        cy.visit(getWebhookBaseUrl() + '/start_oauth');
+
+        // # Click on the allow button
+        cy.findByText('Allow').click();
+
+        // * Exchange succesful
+        cy.findByText('OK').should('exist');
+    });
+
+    it('MM-T648 Part 2 - OAuth 2.0 Zapier tests', () => {
```

Comment:
Can you please combine all this in to one single test. We track the execution of test results from Cypress using the test ID MM-T648 and having multiple tests with same name will cause confusion and overwrite the test results.

Reply:
Regarding this and the next comment: It is complicated to have them in the same test, since we have to run the Aux text in the middle to continue.

Reply:
@larkox can you combine all the 3 tests into a single test? Any reason to have Aux test as a separate test?

Reply:
Cypress does not allow to do cy.visit(x) to different domains inside the same test. Therefore, if we want to perform the oauth login step, we would need to divide it in different tests.

Reply:
I think it's a valid point from @srkgupta but thanks @larkox for clarifying. With that, I'm fine to break the test into 3 steps. I'll update the test case later to match the steps here.